[CI] EsqlSpecIT test {csv-spec:text-embedding.text_embedding with multiple knn queries in fork} failing

**Build Scans:**
- [elasticsearch-intake #37589 / part3](https://gradle-enterprise.elastic.co/s/a7oafk4llkqj4)
- [elasticsearch-pull-request #137468 / part-3](https://gradle-enterprise.elastic.co/s/djtfnzbzy4azm)
- [elasticsearch-intake #37588 / part3](https://gradle-enterprise.elastic.co/s/cmzcdoofgz5ms)
- [elasticsearch-pull-request #137461 / part-3](https://gradle-enterprise.elastic.co/s/pbtbszermewji)
- [elasticsearch-pull-request #137458 / part-3](https://gradle-enterprise.elastic.co/s/j2g3onyr6zezi)
- [elasticsearch-pull-request #137448 / part-3](https://gradle-enterprise.elastic.co/s/g23jgxkgomjpq)
- [elasticsearch-pull-request #137444 / part-3](https://gradle-enterprise.elastic.co/s/j47tb4efmh5f4)
- [elasticsearch-pull-request #137447 / part-3](https://gradle-enterprise.elastic.co/s/jthtdgglu7ulk)
- [elasticsearch-pull-request #137452 / part-3](https://gradle-enterprise.elastic.co/s/jglvibuk3dmvk)
- [elasticsearch-pull-request #137446 / part-3](https://gradle-enterprise.elastic.co/s/5tfjlpnhnekzu)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:single-node:javaRestTest" --tests "org.elasticsearch.xpack.esql.qa.single_node.EsqlSpecIT" -Dtests.method="test {csv-spec:text-embedding.text_embedding scores by stable sort}" -Dtests.seed=3B6FE649D9CBA993 -Dtests.locale=or-Orya-IN -Dtests.timezone=Asia/Ulan_Bator -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.single_node.EsqlSpecIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('test%20%7Bcsv-spec%3Atext-embedding.text_embedding%20with%20multiple%20knn%20queries%20in%20fork%7D'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Data mismatch:
row 0 column 2:a list containing
row 0 column 2:0: expected <0.021739130839705467> but was <0.021761586889624596>
row 1 column 2:a list containing
row 1 column 2:0: expected <0.010989011265337467> but was <0.010983290150761604>
row 2 column 2:a list containing
row 2 column 2:0: expected <1.0> but was <0.9999990463256836>
Actual:
text_field:text                                                       | _fork:keyword | _score:double
all we have to decide is what to do with the time that is given to us | fork1         | 0.021761586889624596
all we have to decide is what to do with the time that is given to us | fork2         | 0.010983290150761604
be excellent to each other                                            | fork1         | 0.9999990463256836
be excellent to each other                                            | fork2         | 0.029411764815449715
live long and prosper                                                 | fork1         | 0.0
[truncated]
```

**Issue Reasons:**
- [main] 3 consecutive failures in test test {csv-spec:text-embedding.text_embedding with multiple knn queries in fork}
- [main] 2 consecutive failures in step part3
- [main] 1 consecutive failures in step part-3
- [main] 2 consecutive failures in pipeline elasticsearch-intake
- [main] 1 consecutive failures in pipeline elasticsearch-pull-request
- [main] 34 failures in test test {csv-spec:text-embedding.text_embedding with multiple knn queries in fork} (24.8% fail rate in 137 executions)
- [main] 8 failures in step part3 (18.6% fail rate in 43 executions)
- [main] 26 failures in step part-3 (28.0% fail rate in 93 executions)
- [main] 8 failures in pipeline elasticsearch-intake (18.6% fail rate in 43 executions)
- [main] 18 failures in pipeline elasticsearch-pull-request (21.2% fail rate in 85 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -238,12 +238,6 @@ tests:
 - class: org.elasticsearch.search.fetch.FetchPhaseDocsIteratorTests
   method: testIterateAsyncVisitsLeavesInDocIdOrder
   issue: https://github.com/elastic/elasticsearch/issues/146078
-- class: org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT
-  method: test {csv-spec:text-embedding.text_embedding with multiple knn queries in fork}
-  issue: https://github.com/elastic/elasticsearch/issues/146091
-- class: org.elasticsearch.xpack.esql.qa.single_node.EsqlSpecForceStoredLoadingIT
-  method: test {csv-spec:text-embedding.text_embedding with multiple knn queries in fork}
-  issue: https://github.com/elastic/elasticsearch/issues/146092
 - class: org.elasticsearch.xpack.security.transport.filter.IpFilteringIntegrationTests
   method: testThatIpFilteringIsIntegratedIntoNettyPipelineViaHttp
   issue: https://github.com/elastic/elasticsearch/issues/146102
@@ -292,42 +286,18 @@ tests:
 - class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
   method: test {p0=indices.resolve_cluster/20_resolve_system_index/Resolve system index via resolve/cluster}
   issue: https://github.com/elastic/elasticsearch/issues/146312
-- class: org.elasticsearch.xpack.esql.CsvIT
-  method: test {csv-spec:text-embedding.text_embedding scores by stable sort}
-  issue: https://github.com/elastic/elasticsearch/issues/146338
-- class: org.elasticsearch.xpack.esql.CsvIT
-  method: test {csv-spec:text-embedding.text_embedding with multiple knn queries in fork}
-  issue: https://github.com/elastic/elasticsearch/issues/146339
-- class: org.elasticsearch.xpack.esql.ccq.MultiClusterSpecIT
-  method: test {csv-spec:text-embedding.text_embedding with multiple knn queries in fork}
-  issue: https://github.com/elastic/elasticsearch/issues/146341
-- class: org.elasticsearch.xpack.esql.ccq.MultiClusterSpecIT
-  method: test {csv-spec:text-embedding.text_embedding scores by stable sort}
-  issue: https://github.com/elastic/elasticsearch/issues/146342
 - class: org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT
   method: test {csv-spec:stats.sumAsSurrogateWithOverflowGroupingRow}
   issue: https://github.com/elastic/elasticsearch/issues/146376
 - class: org.elasticsearch.xpack.esql.qa.single_node.RestEsqlIT
   method: testForceSleepsProfile {SYNC}
   issue: https://github.com/elastic/elasticsearch/issues/146377
-- class: org.elasticsearch.xpack.esql.qa.single_node.EsqlSpecIT
-  method: test {csv-spec:text-embedding.text_embedding scores by stable sort}
-  issue: https://github.com/elastic/elasticsearch/issues/146409
-- class: org.elasticsearch.xpack.esql.qa.single_node.EsqlSpecIT
-  method: test {csv-spec:text-embedding.text_embedding with multiple knn queries in fork}
-  issue: https://github.com/elastic/elasticsearch/issues/146091
 - class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
   method: test {p0=mget/60_realtime_refresh/Realtime Refresh}
   issue: https://github.com/elastic/elasticsearch/issues/146410
-- class: org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT
-  method: test {csv-spec:text-embedding.text_embedding scores by stable sort}
-  issue: https://github.com/elastic/elasticsearch/issues/146409
 - class: org.elasticsearch.xpack.diskbbq.DiskBbqTrialLicenseIT
   method: testDefaultIndexOptionsForDenseVector
   issue: https://github.com/elastic/elasticsearch/issues/146415
-- class: org.elasticsearch.xpack.esql.qa.single_node.EsqlSpecForceStoredLoadingIT
-  method: test {csv-spec:text-embedding.text_embedding scores by stable sort}
-  issue: https://github.com/elastic/elasticsearch/issues/146416
 - class: org.elasticsearch.index.reindex.ReindexClientYamlTestSuiteIT
   method: test {yaml=reindex/10_basic/Response format for version conflict with conflicts=proceed}
   issue: https://github.com/elastic/elasticsearch/issues/146419

--- a/x-pack/plugin/esql/qa/testFixtures/src/main/resources/text-embedding.csv-spec
+++ b/x-pack/plugin/esql/qa/testFixtures/src/main/resources/text-embedding.csv-spec
@@ -92,40 +92,23 @@ required_capability: knn_function_v5
 required_capability: fork_v9
 
 FROM dense_vector_text METADATA _score
-| FORK (EVAL query_embedding = TEXT_EMBEDDING("be excellent to each other", "test_dense_inference") | WHERE KNN(text_embedding_field, query_embedding))
-       (EVAL query_embedding = TEXT_EMBEDDING("live long and prosper", "test_dense_inference") | WHERE KNN(text_embedding_field, query_embedding))
-| SORT _score DESC, _fork ASC
-| LIMIT 10
+| FORK (EVAL query_embedding = TEXT_EMBEDDING("be excellent to each other", "test_dense_inference")
+        | WHERE KNN(text_embedding_field, query_embedding)
+        | SORT _score DESC
+        | LIMIT 10)
+       (EVAL query_embedding = TEXT_EMBEDDING("live long and prosper", "test_dense_inference")
+        | WHERE KNN(text_embedding_field, query_embedding)
+        | SORT _score DESC
+        | LIMIT 10)
+| SORT text_field ASC, _fork ASC
 | KEEP text_field, query_embedding, _fork, _score
 ;
 
 text_field:text                                                       | query_embedding:dense_vector | _fork:keyword | _score:double
-be excellent to each other                                            | [45.0, 55.0, 54.0]           | fork1         | 1.0
-live long and prosper                                                 | [50.0, 57.0, 56.0]           | fork2         | 1.0
-live long and prosper                                                 | [45.0, 55.0, 54.0]           | fork1         | 0.029411764815449715
-be excellent to each other                                            | [50.0, 57.0, 56.0]           | fork2         | 0.029411764815449715
-all we have to decide is what to do with the time that is given to us | [45.0, 55.0, 54.0]           | fork1         | 0.021739130839705467
-all we have to decide is what to do with the time that is given to us | [50.0, 57.0, 56.0]           | fork2         | 0.010989011265337467
-;
-
-text_embedding scores by stable sort
-required_capability: text_embedding_function
-required_capability: dense_vector_field_type_released
-required_capability: knn_function_v5
-required_capability: fork_v9
-
-FROM dense_vector_text METADATA _score
-| FORK (EVAL query_embedding = TEXT_EMBEDDING("be excellent to each other", "test_dense_inference") | WHERE KNN(text_embedding_field, query_embedding))
-       (EVAL query_embedding = TEXT_EMBEDDING("live long and prosper", "test_dense_inference") | WHERE KNN(text_embedding_field, query_embedding))
-| SORT text_field ASC, _fork ASC
-| KEEP text_field, _fork, _score
-;
-
-text_field:text                                                       | _fork:keyword | _score:double
-all we have to decide is what to do with the time that is given to us | fork1         | 0.021739130839705467
-all we have to decide is what to do with the time that is given to us | fork2         | 0.010989011265337467
-be excellent to each other                                            | fork1         | 1.0
-be excellent to each other                                            | fork2         | 0.029411764815449715
-live long and prosper                                                 | fork1         | 0.029411764815449715
-live long and prosper                                                 | fork2         | 1.0
+all we have to decide is what to do with the time that is given to us | [45.0, 55.0, 54.0]           | fork1         | 0.02..0.03
+all we have to decide is what to do with the time that is given to us | [50.0, 57.0, 56.0]           | fork2         | 0.01..0.02
+be excellent to each other                                            | [45.0, 55.0, 54.0]           | fork1         | 0.9..1.0
+be excellent to each other                                            | [50.0, 57.0, 56.0]           | fork2         | 0.02..0.3
+live long and prosper                                                 | [45.0, 55.0, 54.0]           | fork1         | 0.02..0.3
+live long and prosper                                                 | [50.0, 57.0, 56.0]           | fork2         | 0.9..1.0
 ;

